python/tests/backends/test_IQM.py

```diff
@@ -196,6 +200,43 @@ def test_IQM_state_preparation_builder():
     assert assert_close(counts["11"], 0., 2)
 
 
+def test_arbitrary_unitary_synthesis():
+
+    cudaq.register_operation("custom_h",
+                             1. / np.sqrt(2.) * np.array([1, 1, 1, -1]))
+    cudaq.register_operation("custom_x", np.array([0, 1, 1, 0]))
+
+    @cudaq.kernel
+    def basic_x():
+        qubit = cudaq.qubit()
+        custom_x(qubit)
+
+    counts = cudaq.sample(basic_x)
+    counts.dump()
+    # Gives result like { 1:999 0:0 }
```

Comment:
nit: is the default shot count 1000 :) ?
```suggestion
    # Gives result like { 1:1000 0:0 }
```

Reply:
I started with checking that the count for '1' is '1000' and got failures locally. On multiple runs I noticed that the output was `{ 1:999 0:0 }`, rather than `{ 1:1000 0:0 }`,  hence the comment... So instead, I check that count for '0' is '0'. 
I don't know if there is an underlying issue.